R/list-simplify.R

```diff
@@ -0,0 +1,40 @@
+list_simplify <- function(x, simplify = NA, ptype = NULL) {
```

Comment:
I'm neutral on whether or not we should export this. I can see advantages to keeping it internal, but exporting it would make it easier to deprecate `as_vector()` and `simplify()`

Reply:
If we do export, we'll probably need to consider the interface as calling `list_simplify(simplify = FALSE)` would be weird.